### PR TITLE
Update node to 14 and OS to 20.04, M4 and autorest to latest

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: eng/pipelines/build.yml [hunk 1]
```diff
@@ -37,9 +37,9 @@ stages:
           - checkout: self
           - checkout: azure-sdk-tools
           - task: NodeTool@0
-            displayName: "Install Node 13.x"
+            displayName: "Install Node 14.x"
```
Comment (pakrym) [inline eng/pipelines/build.yml:40]: Node 13 is not supported anymore.

File: src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs [hunk 2]
```diff
@@ -541,31 +542,50 @@ private void WriteStatusCodeSwitch(CodeWriter writer, CodeWriterDeclaration mess
...
+                                writer.Line($"var {valueVariable:D} = {message}.ExtractResponseContent();");
+                                value = new Reference(valueVariable.ActualName, responseBody.Type);
+                                break;
+                            case ConstantResponseBody body:
+                                writer.Append($"{returnType} {valueVariable:D} = ");
+                                writer.WriteReferenceOrConstant(body.Value);
+                                writer.Line($";");
+                                value = new Reference(valueVariable.ActualName, responseBody.Type);
+                                break;
+                            case StringResponseBody _:
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs:566]: this branch is new. SignalR somehow suddenly requires a new body type.

File: readme.md [hunk 3]
```diff
@@ -916,10 +916,13 @@ input-file: "swagger-document"
 ```
 </details>
 
+<details>
```
Comment (pakrym) [inline readme.md:919]: Hide config from the main page.

File: src/AutoRest.CSharp.V3/AutoRest/Plugins/CSharpGen.cs [hunk 4]
```diff
@@ -129,6 +129,7 @@ public async Task<bool> Execute(IPluginCommunication autoRest)
 
             if (configuration.SaveInputs)
             {
+                await codeModelTask;
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/AutoRest/Plugins/CSharpGen.cs:132]: make sure we save the codemodel file

**Review by ShivangiReja — APPROVED** [review]

**Pull request merged**